### PR TITLE
Remove Using#map and Using#flatMap

### PR DESCRIPTION
**Pull request by NthPortal**
Resolves scala/bug#11225 by removing the offending methods. Not an ideal solution, but there isn't another one at this point, to my knowledge

### CODE REVIEW

File: test/junit/scala/util/UsingTest.scala [hunk 1]
```diff
@@ -230,12 +230,12 @@ class UsingTest {
...
-      _ <- Using(new ExceptionResource)
-      _ <- Using(new ErrorResource)
-    } yield {
-      throw new UsingException("nested `Using`")
-    }
+    val scala.util.Failure(usingException) =
+      Using(new ExceptionResource) { _ =>
+        Using(new ErrorResource) { _ =>
+          throw new UsingException("nested `Using`")
+        }
```
Comment (NthPortal) [inline test/junit/scala/util/UsingTest.scala]: missing `.get` here, which is exactly the reason why `map` and `flatMap` existed in the first place. I'm not happy with this solution at all.

**Comment by SethTisue** [comment]
Can you rebase and see if the new Xss setting fixes the Travis build?

**Comment by NthPortal** [comment]
Sure

**Comment by lrytz** [comment]
[Sadly,](https://github.com/scala/scala/pull/7381#pullrequestreview-170979080) this really makes `Using` less convenient (or not?). I have a tendency in favor of a solution that delays computation and requires an explicit `use()` / `run()`.

**Comment by julienrf** [comment]
It seems that scala-arm [does support for comprehensions](https://github.com/jsuereth/scala-arm#examples). Is it possible to learn from this project?

**Comment by javax-swing** [comment]
Looks like that scala-arm project uses the lazy approach during map and flatMap. And has terminal methods to get the value from it and use the resource. I for one prefer this to not having map/flatMap

**Comment by NthPortal** [comment]
I will look into scala-arm. I have already looked at better-files, which also supports for comprehensions, but it has the same issue with filters and assignments.

**Comment by dwijnand** [comment]
Check out https://github.com/sbt/io/blob/v1.2.2/io/src/main/scala/sbt/io/Using.scala too.

**Comment by NthPortal** [comment]
@dwijnand I love how the implementation is `open(src)` 😉

**Comment by NthPortal** [comment]
Closing this in favour of #7415, as this solution isn't a very good one.

**Pull request closed without merging**